### PR TITLE
Fix depwarns on julia 0.6 & Optim 0.7.5

### PR DESCRIPTION
**Pull request by yuyichao**
A Optim.jl PR is also ready and will be submitted after https://github.com/JuliaLang/Compat.jl/pull/326 is tagged (Hopefully tomorrow).

### CODE REVIEW

File: test/curve_fit.jl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 let
     # fitting noisy data to an exponential model
-    model(x, p) = p[1]*exp(-x.*p[2])
+    model(x, p) = @compat p[1] .* exp.((-).(x) .* p[2])
```
Comment (blakejohnson) [inline test/curve_fit.jl]: `-x` really doesn't work anymore??

Comment (blakejohnson) [inline test/curve_fit.jl]: I think you've overdone it here... it should be sufficient to just vectorize the call to `exp` so it is:
```
p[1] * exp.(-x .* p[2])
```

Comment (yuyichao) [inline test/curve_fit.jl]: This just make the whole expression a single broadcast.

Comment (blakejohnson) [inline test/curve_fit.jl]: Yes, but it does it at the expense of readability. We've made various choices in LsqFit to sacrifice a small amount of performance in the name of readability. In this case, I find `(-).(x)` to be extraordinarily obtuse. So, please... let's use the easier to read version.

Comment (yuyichao) [inline test/curve_fit.jl]: I'd always make the opposite sacrifice for a 2x reduction of allocation especially for non-public API but I guess this is only the test so I don't care. I changed it to the slower version and left TODO's for how to update it in the future.

Comment (blakejohnson) [inline test/curve_fit.jl]: Thanks.

**Comment by tkelman** [comment]
Weren't dot calls in Compat 0.7.15? https://github.com/JuliaLang/Compat.jl/commit/03fd3eb662fbf766ad4b8aaaccebabe10cdcd059

**Comment by yuyichao** [comment]
I usually use https://github.com/JuliaLang/Compat.jl/commit/fbf0173bcfc759de13c063d8e243cb6dd0bca847 in order to not having to check if the case was supported correctly by 0.4 broadcast.

**Comment by tkelman** [comment]
do the tests fail with earlier versions?

**Comment by yuyichao** [comment]
I don't have an old version so I don't know. It's a behavioral change though so I see 0.9.1 as the first version this is properly implemented.

**Pull request merged**